src/api/routers/usersRouter.ts

```diff
@@ -1,153 +1,30 @@
-import { ParticipantType } from '../entities/ParticipantType';
-import { UserRole } from '../entities/User';
-import { getLoggers, getTraceId } from '../helpers/loggingHelpers';
-import { mapClientTypeToParticipantType } from '../helpers/siteConvertingHelpers';
-import { getKcAdminClient } from '../keycloakAdminClient';
-import { getSite, getSiteList } from '../services/adminServiceClient';
-import {
-  assignClientRoleToUser,
-  deleteUserByEmail,
-  queryUsersByEmail,
-  sendInviteEmail,
-  updateUserProfile,
-} from '../services/kcUsersService';
-import {
-  enrichCurrentUser,
-  enrichUserWithIsApprover,
-  enrichWithUserFromParams,
-  UserRequest,
-} from '../services/usersService';
+import { UserController } from '../controllers/userController';
+import {  UserService } from '../services/userService';
+import { enrichCurrentUser, enrichWithUserFromParams } from '../services/usersService';
 
-export function createUsersRouter() {
+const createUsersRouter = () => {
   const usersRouter = express.Router();
-  usersRouter.use('/current', enrichCurrentUser);
-  usersRouter.get('/current', async (req: UserRequest, res) => {
-    const userWithIsApprover = await enrichUserWithIsApprover(req.user!);
-    return res.json(userWithIsApprover);
-  });
-
-  usersRouter.put('/current/acceptTerms', async (req: UserRequest, res) => {
-    if (!req.user?.participantId) {
-      return res.status(403).json({
-        message: 'Unauthorized. You do not have the necessary permissions.',
-        errorHash: req.headers.traceId,
-      });
-    }
-
-    const participant = await Participant.query().findById(req.user.participantId!);
-    if (!participant || participant.status !== ParticipantStatus.Approved) {
-      return res.status(403).json({
-        message: 'Unauthorized. You do not have the necessary permissions.',
-        errorHash: req.headers.traceId,
-      });
-    }
-
-    const kcAdminClient = await getKcAdminClient();
-    const promises = [
-      req.user!.$query().patch({ acceptedTerms: true }),
-      assignClientRoleToUser(kcAdminClient, req.user?.email!, 'api-participant-member'),
-    ];
-    await Promise.all(promises);
-    return res.sendStatus(200);
-  });
-
-  usersRouter.get('/current/participant', async (req: UserRequest, res) => {
-    const currentParticipant = await Participant.query().findOne({ id: req.user!.participantId });
-
-    const currentSite =
-      currentParticipant?.siteId === undefined
-        ? undefined
-        : await getSite(currentParticipant?.siteId);
-
-    const allParticipantTypes = await ParticipantType.query();
-    const result = {
-      ...currentParticipant,
-      types: mapClientTypeToParticipantType(currentSite?.clientTypes || [], allParticipantTypes),
-    };
-    return res.status(200).json(result);
-  });
+  const userController = new UserController(new UserService());
 
+  usersRouter.use('/current', enrichCurrentUser);
```

Comment:
We'll have to be careful about the orderings here (because some of them are performing auth checks), and ... if we don't have to use different middleware for different routes, can we just rely on the annotations to register the routers with the right HTTP verbs?

One option would be to use a different controller class depending on which middleware (e.g. enrichCurrentUser, enrichWithUserFromParams, etc) that route needs. I.e. there could be a CurrentUserController and a ParamsUserController.